### PR TITLE
docs: clean up multilabel soft margin loss args

### LINKED ISSUE #67969: [docs] F.multilabel_soft_margin_loss does not list arg reduction, but lists outdated size_average
https://pytorch.org/docs/stable/generated/torch.nn.functional.multilabel_soft_margin_loss.html?highlight=multilabel_soft_margin_loss#torch.nn.functional.multilabel_soft_margin_loss

Probably true for other functional loss doc pages

### PR DESCRIPTION
**Pull request by deepujain**
Fixes #67969

Summary:
- Stop listing the deprecated `size_average` and `reduce` arguments in `torch.nn.functional.multilabel_soft_margin_loss` docs.
- Document the existing `weight` argument and fix the return description typo.

Test Plan:
- `python3 -m py_compile torch/nn/functional.py`
- `git diff --check`

### CODE REVIEW

**Comment by deepujain** [comment]
Not asking for review yet; I checked for rendered-doc evidence and only see the fork-safe checks in the visible rollup, not a `docs-build` artifact. The docs-preview page for this generated functional doc is not available yet.

Could a maintainer approve/run docs-build when convenient? Once the preview or artifact exists, I can add a rendered screenshot for `torch.nn.functional.multilabel_soft_margin_loss`.
